### PR TITLE
Use response statuscode when setting loglevel

### PR DESCRIPTION
**Pull request by superseb**
https://github.com/rancher/rancher/issues/24994

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -79,15 +81,19 @@ func run(c *cli.Context) error {
...
 	if err != nil {
 		return err
 	}
+	if response.StatusCode != http.StatusOK {
+		body, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return err
+		}
+		return errors.New(strings.TrimRight(string(body), "\n"))
+	}
```
Comment (cbron) [inline main.go:96]: So the change here is to not limit to 3 levels, but allow any input and see if setting that errors out ? 
I'm unfamiliar with context here, but do we want to print entire response body ? Depending on request couldn't it be a bunch of random junk.

Comment (superseb) [reply main.go:96]: We have a CLI (this) and a logserver (relevant code here https://github.com/rancher/rancher/blob/master/pkg/logserver/server.go#L57), first we checked the loglevel in the CLI but then we want to add TRACE as a new loglevel and I saw that the server (see linked code) actually already does the validation but we don't check that (`err` from http client does not have anything to do with status code). Let me know if this clears things up. I can add more context if needed.

**Review by daxmc99 — APPROVED** [review]
Glad to see another log level get added

File: main.go (hunk 1 above)
Comment (cbron) [reply main.go:96]: Ahh, makes more sense. I suppose if we get a random http error you'd want that printed here anyway.

**Review by cbron — APPROVED** [review]

**Pull request merged**